Description: fix RC bug 640978: "more verbose iproute makes vpnc fail connecting"
Author: Florian Schlichting <fschlich@zedat.fu-berlin.de>
Forwarded: no

diff --git a/vpnc-script.in b/vpnc-script.in
index 59750d0..bd4f120 100755
--- a/vpnc-script.in
+++ b/vpnc-script.in
@@ -116,7 +116,8 @@ destroy_tun_device() {
 
 if [ -n "$IPROUTE" ]; then
 	fix_ip_get_output () {
-		sed 's/cache//;s/metric \?[0-9]\+ [0-9]\+//g;s/hoplimit [0-9]\+//g'
+        sed -e 's/ /\n/g' | \
+            sed -ne '1p;/via/{N;p};/dev/{N;p};/src/{N;p};/mtu/{N;p}'
 	}
 
 	set_vpngateway_route() {
